internal/analytics/ring_buffer.go

```diff
@@ -0,0 +1,124 @@
+package analytics
+
+import "sync"
+
+// EventCollector is a non-blocking analytics producer API.
+type EventCollector interface {
+	// TryAdd attempts to enqueue the event. Returns true if enqueued, false if dropped.
+	TryAdd(interface{}) bool
+}
+
+// RingBuffer provides bounded, non-blocking storage for analytics events.
+// Writes never block; when full, events are dropped per policy.
+type RingBuffer struct {
```

Comment:
Replace custom ring buffer with simple capped size array. If array is full, just drop new events